### PR TITLE
Support Sparse transforms in Simulator.get_nengo_params

### LINKED ISSUE #148: Freeze_params not working on connection to conv layer
Minimal example: 
```
inp = tf.keras.Input(shape=(28, 28, 1))
# convolutional layers
conv0 = tf.keras.layers.Conv2D(
    filters=32,
    kernel_size=3,
    activation=tf.nn.relu,
)(inp)

model = tf.keras.Model(inputs=inp, outputs=conv0)  
converter = nengo_dl.Converter(model)

with nengo_dl.Simulator(converter.net) as sim:
    sim.freeze_params(converter.net)
```
gives

```
Build finished in 0:00:00                                                      
Optimization finished in 0:00:00                                               
Construction finished in 0:00:00                                               
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
<ipython-input-3-04830e9955ff> in <module>
     11 
     12 with nengo_dl.Simulator(converter.net) as sim:
---> 13     sim.freeze_params(converter.net)

~/Dropbox/code/nengo/nengo/utils/magic.py in __call__(self, *args, **kwargs)
    179                 return self.wrapper(wrapped, instance, args, kwargs)
    180             else:
--> 181                 return self.wrapper(self.__wrapped__, self.instance, args, kwargs)
    182         else:
    183             instance = getattr(self.__wrapped__, "__self__", None)

~/Dropbox/code/nengo_dl/nengo_dl/simulator.py in require_open(wrapped, instance, args, kwargs)
     64         )
     65 
---> 66     return wrapped(*args, **kwargs)
     67 
     68 

~/Dropbox/code/nengo_dl/nengo_dl/simulator.py in freeze_params(self, objs)
   1330             for o, params in zip(todo, self.get_nengo_params(todo)):
   1331                 for k, v in params.items():
-> 1332                     setattr(o, k, v)
   1333 
   1334     def get_nengo_params(self, nengo_objs, as_dict=False):

~/Dropbox/code/nengo/nengo/base.py in __setattr__(self, name, val)
    106                 SyntaxWarning,
    107             )
--> 108         super().__setattr__(name, val)
    109 
    110     def __str__(self):

~/Dropbox/code/nengo/nengo/config.py in __setattr__(self, name, val)
    490             except ValidationError:
    491                 exc_info = sys.exc_info()
--> 492                 raise exc_info[1].with_traceback(None)
    493         else:
    494             super().__setattr__(name, val)

ValidationError: init: Shape of initial value (21632,) does not match expected shape (21632, 32)

```

### PR DESCRIPTION
**Pull request by drasmuss**
This functionality (and `Simulator.freeze_params`, which depends on it) had not been updated after the introduction of Sparse transforms.

Fixes #148

### CODE REVIEW

File: nengo_dl/tests/test_simulator.py [hunk 1]
```diff
@@ -1323,6 +1323,57 @@ def test_get_nengo_params(Simulator, seed):
...
+                transform=nengo.Sparse(
+                    (10, 10), indices=[[i] * 2 for i in range(10)], init=data
+                ),
+            )
+
+        p0 = nengo.Probe(node1)
+
+    with Simulator(net) as sim0:
+        params = sim0.get_nengo_params(conn)
+        assert np.allclose(params["transform"].init.data, data)
```
Comment (hunse) [inline nengo_dl/tests/test_simulator.py:1361]: Will this work if Scipy is not installed? In that case, what will `params["transform"].init` be? I'm assuming a Numpy array, but right now when I try without Scipy I'm getting `Aborted (core dumped)`. I'll look into it a bit more. Is this expected to work without Scipy, or does NengoDL only support sparse with Scipy (in which case, we should mark this whole test as requiring Scipy)?

File: nengo_dl/compat.py [hunk 2]
```diff
@@ -134,6 +134,8 @@ def _conform_to_reference_input(self, tensor, ref_input):
 linalg_onenormest.aslinearoperator = linalg_interface.aslinearoperator
 
 if LooseVersion(nengo.__version__) < "3.1.0":
+    NoTransform = None
```
[the remark this answers is not in the document]
Comment (drasmuss) [reply nengo_dl/compat.py:137]: Ah yeah, `NoTransform = type(None)` sounds good to me.

File: nengo_dl/tests/test_simulator.py (hunk 1 above)
Comment (drasmuss) [reply nengo_dl/tests/test_simulator.py:1361]: This test should be skipped if scipy isn't installed (in the `use_scipy=True` condition). In the `use_scipy=False` condition, `.init.data` will be `SparseMatrix.data`.

Comment (hunse) [reply nengo_dl/tests/test_simulator.py:1361]: I forgot we had the `SparseMatrix`. Anyway, I reinstalled and uninstalled Scipy and the weird error disappeared, so who knows? But it seems good now.

**Pull request merged**